tests/unit_tests/get_payments.cpp

```diff
@@ -0,0 +1,72 @@
+//    conditions and the following disclaimer.
+// 
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list
+//    of conditions and the following disclaimer in the documentation and/or other
+//    materials provided with the distribution.
+// 
+// 3. Neither the name of the copyright holder nor the names of its contributors may be
+//    used to endorse or promote products derived from this software without specific
+//    prior written permission.
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// FIXME: move this into a full wallet2 unit test suite, if possible
+
+#include "gtest/gtest.h"
+
+#include "wallet/wallet2.h"
+#include <string>
+
+
+TEST(GetPayments, Success)
+{
+	tools::wallet2 m_wallet;
+        const std::string password = "testpass";
+        crypto::secret_key recovery_key = crypto::secret_key();
+        m_wallet.generate("", password, recovery_key, true, false);
+	std::list<std::pair<crypto::hash, tools::wallet2::payment_details>> matched_payments;
+
+	uint64_t min_height = 0;
+	uint64_t max_height = (uint64_t)-1;
+
+        crypto::hash payment_id;
+	tools::wallet2::payment_details payment;
+	payment.m_block_height = 7;
+        tools::wallet2::payment_container actual_payments ( {{payment_id, payment}} ); 
+
+        uint64_t block_height_that_has_transaction = payment.m_block_height;
+
+
+        ASSERT_EQ(0, matched_payments.size());
+
+        // generally, the function works
+        matched_payments.clear();
+	m_wallet.get_payments_min_height_inclusive(matched_payments, actual_payments, min_height, max_height);
+        ASSERT_EQ(1, matched_payments.size());
+
+        // max_height is inclusive
+        matched_payments.clear();
+	m_wallet.get_payments_min_height_inclusive(matched_payments, actual_payments, min_height, block_height_that_has_transaction);
+        ASSERT_EQ(1, matched_payments.size());
+
+        // min_height is inclusive
+        matched_payments.clear();
+	m_wallet.get_payments_min_height_inclusive(matched_payments, actual_payments, block_height_that_has_transaction, block_height_that_has_transaction);
+        ASSERT_EQ(1, matched_payments.size());
+}
```

Comment:
Can you add more of them ? At the moment it's just one, does not test much.

Reply:
This should do: e65fd7f00130ba4eea75c680b41b9409335e3f8a

I cleaned the structure for the tests and added several new variants, all of which are now also implemented for the get_payments_out function.

Reply:
There's just one payment in the list. I'd expect a test to test when there are no payments, more than one payment per height, payments both in and out of the range you query, etc.